sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/challenge_auth_policy.py

```diff
@@ -40,6 +42,26 @@ def _update_challenge(request, challenger):
         ChallengeCache.set_challenge_for_url(request.http_request.url, challenge)
         return challenge
 
+    @staticmethod
+    def _get_challenge_request(request):
+        # type: (PipelineRequest) -> PipelineRequest
+
+        # The challenge request is intended to provoke an authentication challenge from Key Vault, to learn how the
+        # service request should be authenticated. It should be identical to the service request but with no body.
+        # (Sending the body would be a waste because the challenge request is unauthorized--it's certain to fail.)
+        challenge_request = HttpRequest(
+            request.http_request.method, request.http_request.url, headers=request.http_request.headers
+        )
+
+        if request.http_request.body:
+            # challenge_request has service_request's headers, including Content-Length, if any
+            challenge_request.headers["Content-Length"] = "0"
```

Comment:
Why not just do this unconditionally every time?

Reply:
I didn't want to send a GET with Content-Length. However, Key Vault apparently doesn't care provided the value is 0.